src/rust/lib/graph-editor/src/component/node/port/output.rs

```diff
@@ -0,0 +1,682 @@
+        let full_shape_border_length = &center_segment_length + &corner_segment_length * 2.0;
+
+        let position_relative = index / port_num;
+        let crop_segment_pos  = &position_relative * &full_shape_border_length + position_offset;
+
+        let crop_plane_pos_relative = calculate_crop_plane_position_relative_to_center_segment(
+            &full_shape_border_length, &corner_segment_length, &crop_segment_pos);
+        let crop_plane_pos          = crop_plane_pos_relative * &center_segment_length
+                                      + corner_radius;
+
+        let plane_rotation_angle = compute_border_perpendicular_angle(&full_shape_border_length,
+                                                                      &corner_segment_length,
+                                                                      &crop_segment_pos);
+        let plane_shape_offset = Var::<Distance<Pixels>>::from(&crop_plane_pos - width * 0.5);
+        let crop_shape         = HalfPlane();
+        let crop_shape         = crop_shape.rotate(plane_rotation_angle);
+        let crop_shape         = crop_shape.translate_x(plane_shape_offset);
+
+        crop_shape.into()
+    }
+
+    ensogl::define_shape_system! {
+        (style:Style, grow:f32, index:f32, port_num:f32, opacity:f32, padding_left:f32,
+        padding_right:f32) {
+            let overall_width  : Var<Distance<Pixels>> = "input_size.x".into();
+            let overall_height : Var<Distance<Pixels>> = "input_size.y".into();
+
+            let base_shape_data = port_base_shape_data(&overall_width, &overall_height, &grow);
+            let BaseShapeData{ port_area,hover_area,radius,width } = base_shape_data;
+
+            let left_shape_crop  = compute_crop_plane(&index,
+                                                      &port_num,
+                                                      &width.clone().into(),
+                                                      &radius.clone().into(),
+                                                      &padding_left);
+            let right_shape_crop = compute_crop_plane(&(Var::<f32>::from(1.0) + &index),
+                                                      &port_num,
+                                                      &width.clone().into(),
+                                                      &radius.clone().into(),
+                                                      &padding_right);
+
+            let port_area  = port_area.difference(&left_shape_crop);
+            let port_area  = port_area.intersection(&right_shape_crop);
+
+            let left_hover_crop  = compute_crop_plane(&index, &port_num, &width.clone().into(),
+                                                      &radius.clone().into(), &0.0.into());
+            let right_hover_crop = compute_crop_plane(&(Var::<f32>::from(1.0) + &index), &port_num,
+                                                      &width.into(), &radius.into(), &0.0.into());
+
```

Comment:
What are `left_shape_crop`, `right_shape_crop`, left_hover_crop`, and `right_hover_crop`? This shader looks very, very heavy and I believe it could be simplified a lot. Lets have a chat about it tomorrow.